src/core_plugins/kibana/public/management/sections/indices/create_index_pattern/create_index_pattern.js

```diff
@@ -52,97 +187,108 @@ uiModules.get('apps/management')
-        if (id) {
-          refreshKibanaIndex().then(function () {
-            if (!config.get('defaultIndex')) {
-              config.set('defaultIndex', indexPattern.id);
-            }
-            indexPatterns.cache.clear(indexPattern.id);
-            kbnUrl.change('/management/kibana/indices/' + indexPattern.id);
-          });
+      refreshKibanaIndex().then(function () {
+        if (!config.get('defaultIndex')) {
+          config.set('defaultIndex', id);
         }
-      });
 
-      // refreshFields calls save() after a successfull fetch, no need to save again
-      // .then(function () { indexPattern.save(); })
-    })
-    .catch(function (err) {
+        indexPatterns.cache.clear(id);
+        kbnUrl.change(`/management/kibana/indices/${id}`);
+      });
+    }).catch(err => {
       if (err instanceof IndexPatternMissingIndices) {
-        notify.error($translate.instant('KIBANA-NO_INDICES_MATCHING_PATTERN'));
+        return notify.error($translate.instant('KIBANA-NO_INDICES_MATCHING_PATTERN'));
       }
-      else notify.fatal(err);
+
+      notify.fatal(err);
     });
   };
 
-
   $scope.$watchMulti([
-    'index.isTimeBased',
-    'index.nameIsPattern',
-    'index.nameInterval.name'
+    'newIndexPattern.isTimeBased',
+    'newIndexPattern.nameIsPattern',
+    'newIndexPattern.nameInterval.name'
   ], function (newVal, oldVal) {
     const isTimeBased = newVal[0];
     const nameIsPattern = newVal[1];
-    const newDefault = getPatternDefault(newVal[2]);
-    const oldDefault = getPatternDefault(oldVal[2]);
+    const newDefault = getDefaultPatternForInterval(newVal[2]);
+    const oldDefault = getDefaultPatternForInterval(oldVal[2]);
 
-    if (index.name === oldDefault) {
-      index.name = newDefault;
+    if ($scope.newIndexPattern.name === oldDefault) {
+      $scope.newIndexPattern.name = newDefault;
     }
 
     if (!isTimeBased) {
-      index.nameIsPattern = false;
+      $scope.newIndexPattern.nameIsPattern = false;
     }
 
     if (!nameIsPattern) {
-      delete index.nameInterval;
-      delete index.timeField;
+      delete $scope.newIndexPattern.nameInterval;
+      delete $scope.newIndexPattern.timeField;
     } else {
-      index.nameInterval = index.nameInterval || intervals.byName.days;
-      index.name = index.name || getPatternDefault(index.nameInterval);
+      $scope.newIndexPattern.nameInterval = $scope.newIndexPattern.nameInterval || intervals.byName.days;
+      $scope.newIndexPattern.name = $scope.newIndexPattern.name || getDefaultPatternForInterval($scope.newIndexPattern.nameInterval);
     }
   });
 
   $scope.moreSamples = function (andUpdate) {
-    index.sampleCount += 5;
+    $scope.ctrl.sampleCount += 5;
     if (andUpdate) updateSamples();
   };
 
   $scope.$watchMulti([
-    'index.name',
-    'index.nameInterval'
+    'newIndexPattern.name',
+    'newIndexPattern.nameInterval'
   ], function (newVal, oldVal) {
+    function promiseMatch(lastPromise, cb) {
```

Comment:
In the original file this function was extracted, which confused me. I read its name and implementation, but I still couldn't figure out how it fit into the big picture of "index pattern creation".

After grepping and finding that it was only called from within this handler, I realized that it was just a helper function, meant to DRY up some duplicate logic here.

Since I don't really think it's a meaningful abstraction in the greater context of "index pattern creation", and it's pretty tightly-coupled to this handler's implementation, I felt it made more sense to keep it scoped to just the handler.